src/dev/build/tasks/nodejs_modules/clean_client_modules_on_dll_task.js

```diff
@@ -49,9 +49,18 @@ export const CleanClientModulesOnDLLTask = {
       `${baseDir}/src/plugins/*/server/index.js`,
       `!${baseDir}/src/plugins/**/public`
     ]);
+    const discoveredNewPlatformXpackPlugins = await globby([
+      `${baseDir}/x-pack/plugins/*/server/index.js`,
```

Comment:
any extra context on server files that need to be discovered?

Reply:
We had the need to discover dependencies from any server files in order to not delete dependencies present in the dll also used in the server code. In the past we do not had the need to check into the new platform plugins because there was none, but right now, with more NP plugins being added, we need to check on them too.